Source/Core/DiscIO/Volume.h

```diff
@@ -74,6 +74,7 @@ class Volume
   {
     return INVALID_CERT_CHAIN;
   }
+  virtual std::vector<u8> GetContent(u16 index) const { return {}; }
```

Comment:
I kinda wonder if this should be here, since only WADs (or rather: installed titles) need this.
Then again, the same goes for `GetContentOffsets` below (and potentially others), so we can probably just leave it as-is for now...

Reply:
That would require us to conditionally cast `Volume` to `VolumeWAD` in e.g. `VolumeVerifier`, which also feels pretty ugly. But I agree, having this much stuff in `Volume` that only is implemented in a single subclass isn't great.

Reply:
Yeah, thats the main reason I didn't flag it as something that _must_ be changed for now; since I'm not quite sure how to implement this in a nicer way.
I did have to look twice to notice that `VolumeVerifier::CheckContentIntegrity` is _only_ called for WADs; deleting an unpublished comment about how this would break for non-WADs - so it might be something to consider doing over there first (dunno, like specializing the Verifier etc.) before ripping things out of `Volume` or casting in various places.

Reply:
Yeah, let's leave this as is for now. Currently, tons of Volume member functions only make sense for specific volume types...